TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/Index.cshtml.cs

```diff
@@ -41,8 +46,40 @@ public async Task OnGetAsync()
         Person = await BuildPersonInfoAsync();
         PersonProfessionalStatus = await BuildPersonStatusInfoAsync();
 
-        ShowMergeButton = !HasOpenAlert && Person!.IsActive &&
-            (PersonProfessionalStatus is not PersonProfessionalStatusInfo professionalStatus || !_invalidInductionStatusesForMerge.Contains(professionalStatus.InductionStatus));
+        var contactsMigrated = featureProvider.IsEnabled(FeatureNames.ContactsMigrated);
+
+        var canEditPersonData = (await authorizationService.AuthorizeAsync(User, AuthorizationPolicies.PersonDataEdit))
+            .Succeeded;
+
+        var canEditNonPersonOrAlertData = (await authorizationService.AuthorizeAsync(User, AuthorizationPolicies.NonPersonOrAlertDataEdit))
+            .Succeeded;
+
+        CanChangeDetails =
+            contactsMigrated &&
+            canEditPersonData &&
+            Person.IsActive;
+
+        CanMerge =
+            contactsMigrated &&
+            canEditNonPersonOrAlertData &&
+            !HasOpenAlert &&
+            Person!.IsActive &&
+            (PersonProfessionalStatus is not PersonProfessionalStatusInfo professionalStatus ||
+                !_invalidInductionStatusesForMerge.Contains(professionalStatus.InductionStatus));
+
+        // Person cannot be reactivated if they were deactivated as part of a merge
+        // where they were merged into another Person (i.e. they were the secondary
+        // Person and the other Person was primary)
+        var personWasDeactivatedAsPartOfAMerge = !Person.IsActive &&
```

Comment:
Might be simpler to check `MergedWithPersonId is not null` rather than querying events

Reply:
I missed that in the review! `MergedWithPersonId` should be set on the deactivated record (and point to the retained record).

Reply:
OK I'll leave it as-is for now and add that into the merge process, and update this as part of that change